### PR TITLE
raidboss: add R4S Midnight Sabbath wings and guns info

### PR DESCRIPTION
**Pull request by YuzukiTsuru**
add R4S Midnight Sabbath wings and guns info, When dealing this, The previous ones didn't show the safe zone very visually. Only call:
- Cardinals => Stack With Partner
- Intercards => Spread

In this change, modify to
- Cardinals => Avoid Line + Stack With Partner 
- Intercards => Donut + Spread

It's easier to visual danger zone and find safe areas in the game.

### CODE REVIEW

File: ui/raidboss/data/07-dt/raid/r4s.ts
Comment (wexxlee) [inline ui/raidboss/data/07-dt/raid/r4s.ts:1432]: Was there a reason this wasn't sufficient or accurate?
```
        cardinals: Outputs.cardinals,
        intercards: Outputs.intercards,
```
If there's a translation issue with those, it should probably be fixed in `resources/outputs.ts`, rather than just hard-coding these values.

File: ui/raidboss/data/07-dt/raid/r4s.ts
Comment (wexxlee) [inline ui/raidboss/data/07-dt/raid/r4s.ts:1490]: Same comment here.

File: ui/raidboss/data/07-dt/raid/r4s.ts
Comment (wexxlee) [inline ui/raidboss/data/07-dt/raid/r4s.ts:1466]: Same comment here.

**Review by wexxlee — COMMENTED** [review]
If this does get merged, I think the comments below should probably be addressed.  That said, I'm not sure there's high value in adding 'Donut' or 'Avoid Line' (typo? 'Avoid Lines`?) to the callout, as it doesn't (I don't think?) change how players should position or resolve the mechanic.  And extraneous info in alerts just means more information to read/process, longer tts, etc.

Maybe I'm missing something though, so if anyone else wants to weigh in?

File: ui/raidboss/data/07-dt/raid/r4s.ts
Comment (YuzukiTsuru) [reply ui/raidboss/data/07-dt/raid/r4s.ts:1432]: In the `resources/outputs.ts` cardinals only for `十字` and intercards for `四角`. This translation does not specify whether the AoE are present at the place or whether the player is required to get the place. Besides, since this output may already be used in other triggers, So there is no new changes be made for `resources/outputs.ts`

Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r4s.ts:1432]: Ahh, okay, thank

> In the `resources/outputs.ts` cardinals only for `十字` and intercards for `四角`. This translation does not specify whether the AoE are present at the place or whether the player is required to get the place. Besides, since this output may already be used in other triggers, So there is no new changes be made for `resources/outputs.ts`

Ahh, okay, that makes sense.  Thanks for clarifying.  So just to check (and not questioning the translation, just want to make sure the intent of the original code/callout is clear):  For `cardinals`, the updated output string accurately conveys that cardinals are the safe spot, and you've added that there is an AoE that also needs to be avoided while the player is at the cardinal?   Or the updated string only conveys that there is an AoE at the cardinal that needs to be avoided (and might that cause the player to mistakenly then move to the intercard instead)?

Comment (YuzukiTsuru) [reply ui/raidboss/data/07-dt/raid/r4s.ts:1432]: like the:
- `Cardinals => Avoid Line + Stack With Partner`: `去正点 => 躲避直线 + 与搭档分摊`

First `Cardinals` means you needs to go to cardinals, and `Avoid Line` means a line AoE from intercards is pending, `Stack With Partner` indicate you need to stack with partner.

- `Cardinals => Donut  + Spread`: `去正点 => 月环 + 分散`

First `Cardinals` means you needs to go to cardinals, and `Donut` means there is a Donut in the cardinals position, `Spread` indicate you need to spread.

The first only for safe spot location, the second for which type in the safe spot

In fact, the major changes for this pr is the size of the safe spot of line AoE and Donut is difference, the safe spot of Donut is smaller than Line AoE.

Comment (YuzukiTsuru) [reply ui/raidboss/data/07-dt/raid/r4s.ts:1432]: Maybe 
- `Go Cardinals => Avoid Line + Stack With Partner`
- `Go Intercards => Donut  + Spread`

is better

Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r4s.ts]: > Maybe
> 
> * `Go Cardinals => Avoid Line + Stack With Partner`
> * `Go Intercards => Donut  + Spread`
> 
> is better, which can represent this action

That makes sense to me (and thanks for the clarification, super helpful).  I don't have any other comments, other than some lingering hesitation over the callout being somewhat long and less concise than before.

If it's okay, maybe we can leave this PR up for another day or so to see if anyone else has feelings about that -- if not, I'll approve & merge.
(cc: @valarnin , @JLGarber )

Comment (JLGarber) [reply ui/raidboss/data/07-dt/raid/r4s.ts]: (I'll note here up-front that my static isn't here yet, so while I know how the mechanic works I haven't seen it firsthand.)

I think `Go Intercards + Donut + Spread` is good enough. Personally I would drop "Go", but that's opinion only. The rest of this one is as short as realistically possible without losing information.

`Go Cardinals => Avoid Line + Stack With Partner` might be shortened to `Cardinals + Avoid Line + Partner Stack` perhaps. As before, it's difficult to shorten this further. (I do understand the use of the => arrow separator here, but using only plus signs would make it clear that all three of these actions take place at the same time, instead of implying "go cardinals, then avoid the line and the partner stack".)

Those two notes aside, I think these outputs are as short as reasonably possible. They are long, but not unreasonably so.

Comment (wexxlee) [reply ui/raidboss/data/07-dt/raid/r4s.ts]: > I think `Go Intercards + Donut + Spread` is good enough. Personally I would drop "Go", but that's opinion only. The rest of this one is as short as realistically possible without losing information.
> 
> `Go Cardinals => Avoid Line + Stack With Partner` might be shortened to `Cardinals + Avoid Line + Partner Stack` perhaps. [...]

Those make sense to me.  I think that's probably as short as it can be without dropping actual info.

**Comment by wexxlee** [comment]
If you wanted to, I think you could consider doing something like:
```
        cardinals: {
          ...Outputs.cardinals,
          cn: '去正点',
        },
```
(and so on) so as to address the `cn` translation issue you flagged without requiring i18n PRs for all other languages if they don't have the same issue.  But I'm fine to merge either way once the merge conflict is resolved.

**Comment by YuzukiTsuru** [comment]
> If you wanted to, I think you could consider doing something like:
> 
> ```
>         cardinals: {
>           ...Outputs.cardinals,
>           cn: '去正点',
>         },
> ```
> 
> (and so on) so as to address the `cn` translation issue you flagged without requiring i18n PRs for all other languages if they don't have the same issue. But I'm fine to merge either way once the merge conflict is resolved.

Thank you, learned the usage!

**Comment by YuzukiTsuru** [comment]
I checked the historical commit and found that the outputs commit was matched with RS4 trigger commit. There are no other triggers currently using this Output, so the outputs are modified here.

**Comment by wexxlee** [comment]
> I checked the historical commit and found that the outputs commit was matched with RS4 trigger commit. There are no other triggers currently using this Output, so the outputs are modified here.

Yep, you're completely right -- I added those for r4s (and then completely forgot that I did it 🙃).  LGTM, and thanks for your work on this!

**Review by wexxlee — APPROVED** [review]

**Pull request merged**
